linetools/analysis/voigt.py

```diff
@@ -0,0 +1,203 @@
+
+    v=vin
+    # Voigt function is symmetric, so -v = v
+    if len(np.argwhere(v<0.0)) != 0: v[np.argwhere(v<0.0)] *= -1.0
+    # if a is exactly zero go to 3 for exact expression
+    if (a == 0.0):
+        voigt_prof = np.exp(-(v*v))
+        return voigt_prof
+    # Scale up v for ease with lookup tables
+    v0 = v*10.0
+    n=np.array(v0,dtype=np.int_)
+    voigt_prof = np.zeros(np.size(vin))
+    nl=np.argwhere(n<100)
+    nh=np.argwhere(n>=100)
+    if len(nh) != 0:
+        r=1.0/v[nh]**2
+        voigt_prof[nh] = a*r*oneonsqrtpi*(1.0 + r*(1.5 + r*(3.75 + r*(13.125 + 59.0625*r))) - a*a*r*(1.0 + r*(5.0 +26.25*r)))
+    if len(nl) != 0:
+        v0[nl] = 2*v[nl]*10.0
+        p=np.int_(v0[nl])
+        p1=p+1
+        p2=p+2
+        x=0.5*np.int_(v0[nl])
+        y=x+0.5
+        z=x+1.0
+        v1 = v0[nl] * 0.5
+        voigt_prof[nl] = 2.0*((v1-y)*(v1-z)*(h0[p]+a*(h1[p]+a*(h2[p]+a*h3[p]))) - (v1-x)*(v1-z)*2.0*(h0[p1] + a*(h1[p1]+a*(h2[p1]+a*h3[p1]))) + (v1-x)*(v1-y)*(h0[p2] + a*(h1[p2]+a*(h2[p2]+a*h3[p2]))))
+    del nl, nh
+    return voigt_prof
+
+
+# The primary call
+def voigt_model(iwave, line, fwhm=None, flg_ret=1, debug=False, 
+    skip_wveval=False):
+    '''Generates a Voigt model from a line or list of lines
+    This will run *slowly* for many many lines
+
+    Parameters:
+    ------------
+    wave: Quantity array
+      Observed wavelengths
+    line: AbsLine, List of Absline, or array of parameters
+    skip_wveval: bool, optional [False]
+      Skip wavelength check
+      If False, the median dwave value is calculated and compared against
+      1/10 of the b-value of the input line(s).  
+      If necessary, the wavelenght array is rebinned for the calculation
+      and the final array is rebinned to the original.
+    fwhm: float, optional [None]
+      FWHM for Gaussian smoothing (pixels)
+    flg_ret : int (1)  Byte-wise Flag for return
+      1: vmodel [XSpectrum1D]
+      2: tau
+      4: Absorbed flux [np.ndarray]
```

Comment:
Can these flags be made into strings to make them easier to remember?

Maybe change the keyword to just 'ret' and make the options 'vmodel', 'tau' and 'flux'/